cmake/patches/halide_fp_simplify.patch

```diff
@@ -19,3 +19,30 @@ index 7174974..4193ada 100644
          if (propagate_indeterminate_expression(a, b, op->type, &expr)) {
              return;
          }
+@@ -927,6 +937,26 @@ private:
+         } else {
+             expr = Add::make(a, b);
+         }
++        // The following simplify pass is a copy corresponding to part of LLVM InstCombineAddSub pass,
++        // which assume no constant vector applied.
++        // We apply this pass here so LLVM pass won't be active.
++        // This is a walk-around fix for now.
++        // The real proper fix should change the tvm simplify and vectorize passes register order.
++        if (no_overflow(op->type) && (mod_a && mul_b)) {
++          const Mod* mod_b_a = mul_b ? mul_b->a.as<Mod>() : nullptr;
```

Comment:
Seems mul_b can't be nullptr because you have mul_b checked in the if condition?

Reply:
Right.